app/uk/gov/hmrc/trusts/models/SchemaModels.scala

```diff
@@ -0,0 +1,588 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.trusts.models
+
+
+import org.joda.time.DateTime
+import play.api.libs.json._
+/**
+  * DES API Schema - definitions models below
+  */
+
+case class Registration(matchData: Option[MatchData],
+                        correspondence: Correspondence,
+                        yearsReturns: Option[YearsReturns],
+                        declaration: Declaration,
+                        details: Details,
+                        agentDetails: Option[AgentDetails] = None
+                       )
+
+object Registration {
+  implicit val registrationFormat: Format[Registration] = Json.format[Registration]
+}
+
+
+case class Details(estate: Option[Estate],
+                   trust: Option[Trust])
+
+object Details {
+  implicit val detailsFormat: Format[Details] = Json.format[Details]
+}
+
+case class MatchData(utr: String,
+                     name: String,
+                     postCode: Option[String]
+                    )
+
+object MatchData {
+  implicit val matchDataFormat: Format[MatchData] = Json.format[MatchData]
+}
+
+case class Correspondence(abroadIndicator: Boolean,
+                          name: String,
+                          address: AddressType,
+                          phoneNumber: String)
+
+object Correspondence {
+  implicit val correspondenceFormat : Format[Correspondence] = Json.format[Correspondence]
+
+}
+
+case class YearsReturns(var returns: Option[List[YearReturnType]])
+
+object YearsReturns {
+  implicit val yearsReturnsFormat: Format[YearsReturns] = Json.format[YearsReturns]
+}
+
+case class Assets(monetary: Option[List[AssetMonetaryAmount]],
+                  propertyOrLand: Option[List[PropertyLandType]],
+                  shares: Option[List[SharesType]],
+                  business: Option[List[BusinessAssetType]],
+                  partnerShip: Option[List[PartnershipType]],
+                  other: Option[List[OtherAssetType]])
+
+object Assets {
+  implicit val assetsFormat: Format[Assets] = Json.format[Assets]
+}
+
+case class AssetMonetaryAmount(assetMonetaryAmount: Long)
+
+object AssetMonetaryAmount {
+  implicit val assetMonetaryAmountFormat: Format[AssetMonetaryAmount] = Json.format[AssetMonetaryAmount]
+}
+
+case class Declaration(name: NameType,
+                       address: AddressType)
+
+object Declaration {
+  implicit val declarationFormat: Format[Declaration] = Json.format[Declaration]
+}
+
+case class Trust(
+                  details: TrustDetailsType,
+                  entities: TrustEntitiesType,
+                  assets: Assets)
+
+object Trust {
+  implicit val trustFormat: Format[Trust] = Json.format[Trust]
+}
+
+case class TrustEntitiesType(naturalPerson: Option[List[NaturalPersonType]],
```

Comment:
Do we need to define a `Option[List[NatrualPersonType]]` when a `List[T]` can represent an empty state and we can read as so, even if the key does not exists.

Reply:
This applies to where you have defined `Option[List[_]]` in case classes.

Reply:
it is optional field.